ssl/ssl_ciph.c

```diff
@@ -493,8 +493,47 @@ int ssl_cipher_get_evp_md_mac(SSL_CTX *ctx, const SSL_CIPHER *sslc,
     return 1;
 }
 
+int ssl_cipher_get_evp_cipher_ecb(SSL_CTX *ctx, const SSL_CIPHER *sslc,
+                                  const EVP_CIPHER **enc)
+{
+    int i = ssl_cipher_info_lookup(ssl_cipher_table_cipher, sslc->algorithm_enc);
+
+    if (i == -1) {
+        *enc = NULL;
+    } else {
+        if (i == SSL_ENC_NULL_IDX) {
+            /*
+             * We assume we don't care about this coming from an ENGINE so
+             * just do a normal EVP_CIPHER_fetch instead of
+             * ssl_evp_cipher_fetch()
+             */
+            *enc = EVP_CIPHER_fetch(ctx->libctx, "NULL", ctx->propq);
+        } else {
+            char *ecb_name = NULL;
+
+            if ((sslc->algorithm_enc & SSL_AES128_ANY) != 0) {
+                ecb_name = "AES-128-ECB";
+            } else if ((sslc->algorithm_enc & SSL_AES256_ANY) != 0) {
+                ecb_name = "AES-256-ECB";
+            } else if (ossl_assert((sslc->algorithm_enc & SSL_CHACHA20) != 0)) {
+                ecb_name = "ChaCha20";
+            }
+
+            if (ecb_name != NULL)
+                *enc = EVP_CIPHER_fetch(ctx->libctx, ecb_name, ctx->propq);
+            else
+                return 0;
+        }
+
+        if (*enc == NULL)
+            return 0;
```

Comment:
Should this be outside of this if block?

Reply:
I'm trying to mimic the behaviour of `ssl_cipher_get_evp_cipher()`. I'm not sure if it's correct or not?

Reply:
Hmm, right, this should stay inside.